ca/certificates.go

```diff
@@ -497,17 +533,9 @@ func CreateAndWriteRootCA(rootCN string, paths CertPaths) (RootCA, error) {
 		return RootCA{}, err
 	}
 
-	// Ensure directory exists
-	err = os.MkdirAll(filepath.Dir(paths.Cert), 0755)
-	if err != nil {
-		return RootCA{}, err
-	}
-
-	// Write the Private Key and Certificate to disk, using decent permissions
-	if err := ioutils.AtomicWriteFile(paths.Cert, cert, 0644); err != nil {
-		return RootCA{}, err
-	}
-	if err := ioutils.AtomicWriteFile(paths.Key, key, 0600); err != nil {
+	// TODO (cyli):  for backwards compatibility, we are writing the key to disk
+	// unencrypted.  We may want to just not write the key to disk at all.
```

Comment:
@aaronlehmann I left it in just in case folks want a backup, or otherwise want access to the key.  I'd be happy to remove it if everyone agrees most folks don't depend on this.